db/rdbms/src/main/resources/mapper/VariableMapper.xml

```diff
@@ -9,30 +9,162 @@
+      PROCESS_INSTANCE_KEY,
+      SCOPE_KEY,
+      VAR_NAME,
+      VAR_VALUE,
+    VAR_FULL_VALUE,
+      TENANT_ID,
+    IS_PREVIEW
+    FROM VARIABLE
+    <include refid="io.camunda.db.rdbms.sql.VariableMapper.searchFilter"/>
+    <if test="sort != null and sort.orderings != null and !sort.orderings.isEmpty()">
+      <foreach collection="sort.orderings" open="ORDER BY " separator=", " item="item">
+        <include refid="io.camunda.db.rdbms.sql.VariableMapper.variableSortMapper"/>
+        ${item.order}
+      </foreach>
+    </if>
+    ${paging.after}
+  </select>
+
+  <sql id="searchFilter">
+    WHERE 1 = 1
+    <!-- basic filters -->
+    <if
+      test="filter.processInstanceKeyOperations != null and !filter.processInstanceKeyOperations.isEmpty()">
+      <foreach collection="filter.processInstanceKeyOperations" item="operation">
+        AND PROCESS_INSTANCE_KEY
+        <include refid="io.camunda.db.rdbms.sql.Commons.operationCondition"/>
+      </foreach>
+    </if>
+    <if test="filter.scopeKeyOperations != null and !filter.scopeKeyOperations.isEmpty()">
+      <foreach collection="filter.scopeKeyOperations" item="operation">
+        AND SCOPE_KEY
+        <include refid="io.camunda.db.rdbms.sql.Commons.operationCondition"/>
+      </foreach>
+    </if>
+    <if test="filter.variableKeyOperations != null and !filter.variableKeyOperations.isEmpty()">
+      <foreach collection="filter.variableKeyOperations" item="operation">
+        AND VAR_KEY
+        <include refid="io.camunda.db.rdbms.sql.Commons.operationCondition"/>
+      </foreach>
+    </if>
+    <if test="filter.tenantIds != null and !filter.tenantIds.isEmpty()">
+      AND TENANT_ID IN
+      <foreach collection="filter.tenantIds" item="value" open="(" separator=", " close=")">#{value}</foreach>
+    </if>
+
+    <!-- advanced filters on variable name / values -->
+    <if test="filter.variableOperations != null and !filter.variableOperations.isEmpty()">
+      AND (
+      <foreach collection="filter.variableOperations" index="name" item="operations">
+        (VAR_NAME = #{name}
+        <foreach collection="operations" item="operation">
+          AND
+          <include refid="variableOperationCondition"/>
+        </foreach>
+        )
+        OR
+      </foreach>
+      false )
+    </if>
+
+  </sql>
+
+  <sql id="variableOperationCondition">
+    <choose>
+      <when test="operation.operator.name().equals('EQUALS')">
+        VAR_VALUE = #{operation.value}
+      </when>
+      <when test="operation.operator.name().equals('NOT_EQUALS')">
+        VAR_VALUE != #{operation.value}
+      </when>
+      <when test="operation.operator.name().equals('EXISTS')">
+        VAR_VALUE IS NOT NULL
+      </when>
+      <when test="operation.operator.name().equals('NOT_EXISTS')">
+        VAR_VALUE IS NULL
+      </when>
+      <when test="operation.operator.name().equals('GREATER_THAN')">
+        VAR_LONG &gt; #{operation.value} OR VAR_DOUBLE &gt; #{operation.value}
```

Comment:
❌ These could yield wrong results with the comparison and `or` condition if we use primitives with 0.

I think we need to either:
- Use null instead of 0 for _empty_ values and add null checks here.
- Add another column where we store the type of the variable and use the corresponding value for comparisons. This might be the cleanest solution. You might be able to reuse the `operationCondition` snippet.

Reply:
You are right, I added a type column and use it here for the conditions